src/libYARP_OS/include/yarp/os/Log.h

```diff
@@ -70,7 +70,7 @@ class YARP_OS_API Log
     void info(const char *msg, ...) const YARP_ATTRIBUTE_FORMAT(printf, 2, 3);
     void warning(const char *msg, ...) const YARP_ATTRIBUTE_FORMAT(printf, 2, 3);
     void error(const char *msg, ...) const YARP_ATTRIBUTE_FORMAT(printf, 2, 3);
-    void fatal(const char *msg, ...) const YARP_ATTRIBUTE_FORMAT(printf, 2, 3);
+    void fatal YARP_NORETURN (const char *msg, ...) const YARP_ATTRIBUTE_FORMAT(printf, 2, 3);
```

Comment:
I updated the PR description with the explanation of this placement foundo on Stackoverflow. 
If you don't put it there with clang it doesn't compile.